src/contracts.rs

```diff
@@ -583,3 +607,339 @@ impl WatchOnlySwapCoin {
+        );
+    }
+
+    #[test]
+    fn test_contract_sig_validation() {
+        // First create a funding transaction
+        let secp = Secp256k1::new();
+        let priv_1 =
+            PrivateKey::from_wif("cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy").unwrap();
+        let priv_2 =
+            PrivateKey::from_wif("5JYkZjmN7PVMjJUfJWfRFwtuXTGB439XV6faajeHPAM9Z2PT2R3").unwrap();
+
+        let pub1 = priv_1.public_key(&secp);
+        let pub2 = priv_2.public_key(&secp);
+
+        let funding_outpoint_script =
+            crate::wallet_sync::create_multisig_redeemscript(&pub1, &pub2);
+
+        let funding_spk = Address::p2sh(&funding_outpoint_script, NETWORK).script_pubkey();
+
+        let funding_tx = Transaction {
+            input: vec![TxIn {
+                // random oputpoint
+                previous_output: OutPoint::from_str(
+                    "5df6e0e2761359d30a8275058e299fcc0381534545f55cf43e41983f5d4c9456:42",
+                )
+                .unwrap(),
+                sequence: 0,
+                witness: Vec::new(),
+                script_sig: Script::new(),
+            }],
+            output: vec![TxOut {
+                script_pubkey: funding_spk,
+                value: 2000,
+            }],
+            lock_time: 0,
+            version: 2,
+        };
+
+        // Create the contract transaction spending the funding outpoint
+        let funding_outpoint = OutPoint::new(funding_tx.txid(), 0);
+
+        let contract_script = Script::from(Vec::from_hex("827ca914cdccf6695323f22d061a58c398deba38bba47148876321032e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af0120516721039b6347398505f5ec93826dc61c19f47c66c0283ee9be980e29ce325a0f4679ef000812dabb690fe0fd3768b2757b88ac").unwrap());
+
+        let contract_tx = create_receivers_contract_tx(
+            funding_outpoint,
+            funding_tx.output[0].value,
+            &contract_script,
+        );
+
+        // priv1 signs the contract and verify
+        let sig1 = sign_contract_tx(
+            &contract_tx,
+            &funding_outpoint_script,
+            funding_tx.output[0].value,
+            &priv_1.key,
+        );
+
+        assert_eq!(
+            verify_contract_tx_sig(
+                &contract_tx,
+                &funding_outpoint_script,
+                funding_tx.output[0].value,
+                &pub1,
+                &sig1
+            ),
+            true
+        );
+
+        // priv2 signs the contract and verify
+        let sig2 = sign_contract_tx(
+            &contract_tx,
+            &funding_outpoint_script,
+            funding_tx.output[0].value,
+            &priv_2.key,
+        );
+
+        assert_eq!(
+            verify_contract_tx_sig(
+                &contract_tx,
+                &funding_outpoint_script,
+                funding_tx.output[0].value,
+                &pub2,
+                &sig2
+            ),
+            true
```

Comment:
You should use `assert!(something)` instead of `assert_eq!(something, true)`

Reply:
Done.